core/storage/story/gae_models.py

```diff
@@ -0,0 +1,199 @@
+        version number.
+
+        Args:
+            story_id: str. The id of the story being edited.
+            version: int. The version number of the story after the commit.
+
+        Returns:
+            The commit with the given story id and version number.
+        """
+        return cls.get_by_id('story-%s-%s' % (story_id, version))
+
+    @classmethod
+    def get_all_commits(cls, page_size, urlsafe_start_cursor):
+        """Fetches a list of all the commits sorted by their last updated
+        attribute.
+
+        Args:
+            page_size: int. The maximum number of entities to be returned.
+            urlsafe_start_cursor: str or None. If provided, the list of
+                returned entities starts from this datastore cursor.
+                Otherwise, the returned entities start from the beginning
+                of the full list of entities.
+
+        Returns:
+            3-tuple of (results, cursor, more) as described in fetch_page() at:
+            https://developers.google.com/appengine/docs/python/ndb/queryclass,
+            where:
+                results: List of query results.
+                cursor: str or None. A query cursor pointing to the next
+                    batch of results. If there are no more results, this might
+                    be None.
+                more: bool. If True, there are (probably) more results after
+                    this batch. If False, there are no further results after
+                    this batch.
+        """
+        return cls._fetch_page_sorted_by_last_updated(
+            cls.query(), page_size, urlsafe_start_cursor)
+
+
+class StorySummaryModel(base_models.BaseModel):
+    """Summary model for an Oppia Story.
+
+    This should be used whenever the content blob of the story is not
+    needed (e.g. search results, etc).
+
+    A StorySummaryModel instance stores the following information:
+
+        id, description, language_code, last_updated, created_on, version.
+
+    The key of each instance is the story id.
+    """
+
+    # The title of the story.
+    title = ndb.StringProperty(required=True, indexed=True)
+    # The topic id the story corresponds to.
+    topic = ndb.StringProperty(indexed=False)
+    # The ISO 639-1 code for the language this question is written in.
+    language_code = ndb.StringProperty(required=True, indexed=True)
+    # Time when the story model was last updated (not to be
+    # confused with last_updated, which is the time when the
+    # story *summary* model was last updated).
+    story_model_last_updated = ndb.DateTimeProperty(indexed=True)
+    # Time when the story model was created (not to be confused
+    # with created_on, which is the time when the story *summary*
+    # model was created).
+    story_model_created_on = ndb.DateTimeProperty(indexed=True)
+    # The number of nodes that are part of this story.
+    node_count = ndb.IntegerProperty()
```

Comment:
worth indexing on this? (maybe want to query for short stories, long stories, etc. later)

Reply:
Done.